algosdk/template.py

```diff
@@ -175,6 +175,69 @@ def get_program(self):
         return inject(orig, offsets, values, types)
 
 
+class PeriodicPayment(Template):
+    """
+    PeriodicPayment contract enables creating an account which allows the
+    withdrawal of a fixed amount of assets every fixed number of rounds to a
+    specific Algrorand Address. In addition, the contract allows to add
+    timeout, which after the address can withdraw the rest of the assets.
+
+    Args:
+        receiver (str): address to receive the assets
+        amount (int): amount of assets to transfer at every cycle
+        withdrawing_window (int): the number of blocks in which the user can
+            withdraw the asset once the period start (must be < 1000)
+        period (int): how often the address can withdraw assets (in rounds)
+        fee (int): maximum fee per transaction
+        timeout (int): a round in which the receiver can withdraw the rest of
+            the funds after
+    """
+    def __init__(self, receiver: str, amount: int, withdrawing_window: int,
+                 period: int, max_fee: int, timeout: int):
+        self.lease_value = bytes([random.randint(0, 255) for x in range(
+                                 constants.lease_length)])
+        self.receiver = receiver
+        self.amount = amount
+        self.withdrawing_window = withdrawing_window
+        self.period = period
+        self.max_fee = max_fee
+        self.timeout = timeout
+        
+    def get_program(self):
+        """
+        Return a byte array to be used in LogicSig.
+        """
+        orig = ("ASAHAQoLAAwNDiYCAQYg/ryguxRKWk6ntDikaBrIDmyhBby2B/xWUyXJVp" +
+                "X2ohMxECISMQEjDhAxAiQYJRIQMQQhBDECCBIQMQYoEhAxCTIDEjEHKRIQ" +
+                "MQghBRIQMQkpEjEHMgMSEDECIQYNEDEIJRIQERA=")
+        orig = base64.b64decode(orig)
+        offsets = [4, 5, 7, 8, 9, 12, 15]
+        values = [self.max_fee, self.period, self.withdrawing_window,
+                  self.amount, self.timeout, base64.b64encode(
+                      self.lease_value), self.receiver]
+        types = [int, int, int, int, int, "base64", "address"]
+        return inject(orig, offsets, values, types)
+
+    def get_withdrawal_transaction(self, first_valid, last_valid, gh, fee):
```

Comment:
```
&&
txn FirstValid
int TMPL_PERIOD
%
int 0
==
&&
txn LastValid
int TMPL_DUR
txn FirstValid
+
==
```


This is from the contract. `FirstValid` must be a multiple of `period` and `LastValid` must be equal to `duration + firstValid` so worth checking that and computing `LastValid` ourselves